ui-tests/src/test/java/io/syndesis/qe/steps/CommonSteps.java

```diff
@@ -616,6 +616,22 @@ private void fillAndValidateGmail() {
     }
 
     private void fillAndValidateConcur() {
-        //  TODO: SAP connection validation when issue with concur credentials and oauth callback is resolved
+        assertThat(WebDriverRunner.currentFrameUrl())
+                .containsIgnoringCase("concursolutions");
+        Optional<Account> account = AccountsDirectory.getInstance().getAccount("QE Concur");
+
+        if (account.isPresent()) {
+
+            $(By.id("userid")).shouldBe(visible).sendKeys(account.get().getProperty("userId"));
+            $(By.xpath("//*[@type='submit']")).shouldBe(visible).click();
+            TestUtils.sleepForJenkinsDelayIfHigher(3);
+            $(By.id("password")).shouldBe(visible).sendKeys(account.get().getProperty("password"));
+            $(By.xpath("//*[@type='submit']")).shouldBe(visible).click();
+            //give it time to log in
+            TestUtils.sleepForJenkinsDelayIfHigher(4);
```

Comment:
isn't there any way how to change this from thread.sleep to some condition what is visible when the user is logged in?